research/design-doc/sparse_checkout_asishkumar.md

```diff
@@ -0,0 +1,29 @@
+# KPM sparse checkout
+
+**Author**: Asish Kumar
+
+## Solution
+
+In order to add the feature of sparse-checkout, kpm will add the package specified in the kcl.mod file and later when running `kcl mod run` it will checkout that destination directory which contains that package recursively. 
+
+## User Interface
+
+In order to use a specified package within a repository, the user will have to specify the package during the `kcl mod add` command. For example
+
+```
+kcl mod add --git https://github.com/officialasishkumar/check-kcl.git --commit 831fada36e155c8758f07f293c8267c869af69d3 --package k8s
```

Comment:
Add more details, including what happens if `--package` is not added, since `--package` is an option flag, and updates in `kcl.mod` and `kcl.mod.lock` after running.

Reply:
The feature won't work without a package flag. Should i specify that?

Reply:
Yes, we can not do an incomplete function for the user. You need to consider the user's various situations, and make corresponding preparations. For example, you should at least design that when he does not give this field or gives this field, `kcl mod add` works normally or does not work normally, what does `kcl.mod`, `kcl.mod.lock`, and the corresponding CLI output should look like.

Reply:
I have updated accordingly. Can you please take a look?